### DISCUSSION TITLE
Invoke custom Event Handlers

### DISCUSSION THREAD

**Discussion by rochmirabeau** [YetiForceCompany/YetiForceCRM#15541] [How to] [answered]
I wrote a simple handler for a new module

    class TestHandler 
    {    public function entityBeforeSave(App\EventHandler $eventHandler){
         $recordModel = $eventHandler->getRecordModel();
         $recordModel->set('testfield', "event handler changed me");
        }
    }

to $YETI/Modules/CreatedModule/handlers/CustomModuleHandler.php

I added it to the vtiger_eventhandlers table, set is_active to 1, and added the module to the include_modules field. When I save a record I get a LBL_HANDLER_NOT_FOUND error.

Where do I tell yeti which php file to look for the Event Handler?
What are the proper steps to call newly created Event Handlers?

**Comment by rskrzypczak** [comment, marked as answer]
Maybe the reason is in the class name?
"/modules/CreatedModule/handlers/CustomModuleHandler.php" =>"CreatedModule_CustomModuleHandler_Handler"

**Comment by Akaki-1991** [comment]
> I wrote a simple handler for a new module
> 
> ```
> class TestHandler 
> {    public function entityBeforeSave(App\EventHandler $eventHandler){
>      $recordModel = $eventHandler->getRecordModel();
>      $recordModel->set('testfield', "event handler changed me");
>     }
> }
> ```
> 
> to $YETI/Modules/CreatedModule/handlers/CustomModuleHandler.php
> 
> I added it to the vtiger_eventhandlers table, set is_active to 1, and added the module to the include_modules field. When I save a record I get a LBL_HANDLER_NOT_FOUND error.
> 
> Where do I tell yeti which php file to look for the Event Handler? What are the proper steps to call newly created Event Handlers?

hello, How did you make this task, can you explain?
